types/state_transaction.go

```diff
@@ -0,0 +1,137 @@
+package types
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/umbracle/ethgo"
+	"github.com/umbracle/ethgo/abi"
+)
+
+var (
+	StateSyncABIType = abi.MustNewType(
+		"tuple(tuple(uint256 id, address sender, address receiver, bytes data))")
+
+	ExecuteStateSyncABIMethod, _ = abi.NewMethod("function execute(" +
+		"bytes32[] proof, " +
+		"tuple(uint256 id, address sender, address receiver, bytes data) stateSync)")
+)
+
+const (
+	abiMethodIDLength = 4
+)
+
+// StateSyncEvent is a bridge event from the rootchain
+type StateSyncEvent struct {
+	// ID is the decoded 'index' field from the event
+	ID uint64
+	// Sender is the decoded 'sender' field from the event
+	Sender ethgo.Address
+	// Receiver is the decoded 'receiver' field from the event
+	Receiver ethgo.Address
+	// Data is the decoded 'data' field from the event
+	Data []byte
+}
+
+// ToMap converts StateSyncEvent to map
+func (sse *StateSyncEvent) ToMap() map[string]interface{} {
+	return map[string]interface{}{
+		"id":       sse.ID,
+		"sender":   sse.Sender,
+		"receiver": sse.Receiver,
+		"data":     sse.Data,
+	}
+}
+
+// ToABI converts StateSyncEvent to ABI
+func (sse *StateSyncEvent) ToABI() ([]byte, error) {
+	ma := make([]map[string]interface{}, 1)
+	ma[0] = sse.ToMap()
+
+	return StateSyncABIType.Encode(ma)
```

Comment:
Nitpick, maybe try this way?

```suggestion
	return StateSyncABIType.Encode([]interface{}{sse.ToMap()})
```

Reply:
Yep, it is better to encode with array instead of map 0 thought both should work.

Reply:
Much better, done in https://github.com/0xPolygon/polygon-edge/pull/1013/commits/49d9ac7c63011e162ac4fa9b2162401a8a105bbf and finished in https://github.com/0xPolygon/polygon-edge/pull/1013/commits/18ad7ad813b4507514a0b9cea34e2d4f5904690f